### PR TITLE
Client manage relay reputation

### PR DESCRIPTION
**Pull request by drortirosh**
Issue #75

### CODE REVIEW

File: src/js/relayclient/ServerHelper.js [hunk 1]
```diff
@@ -107,24 +111,66 @@ class ActiveRelayPinger {
...
+
         this.httpSend = httpSend
         this.verbose = verbose
-        
+        this.failedRelays = failedRelays
+        this.relayTimeoutGrace = relayTimeoutGrace || DEFAULT_RELAY_TIMEOUT_GRACE_SEC
+
+        this.addScoreRandomness = addScoreRandomness || Math.random
+
+        this.calculateRelayScore = calculateRelayScore || this.defaultCalculateRelayScore.bind(this)
```
Comment (alex-forshtat-tbk) [inline src/js/relayclient/ServerHelper.js]: Why need to bind one but not bind the other?

File: src/js/relayclient/ServerHelper.js [hunk 2]
```diff
@@ -107,24 +111,66 @@ class ActiveRelayPinger {
...
+
+        //default filter: either calculateRelayScore didn't set "score" field,
+        // or if unstakeDelay is below min, or if stake is below min.
         this.relayFilter = relayFilter || ((relay) => (
```
Comment (alex-forshtat-tbk) [inline src/js/relayclient/ServerHelper.js]: Could you please make it a function, just as "calculateRelayScore", for consistency and stuff?

File: src/js/relayclient/ServerHelper.js [hunk 3]
```diff
@@ -172,20 +218,22 @@ class ServerHelper {
...
                 let args = event.returnValues
-                activeRelays[args.relay] = {
+                let relay = {
                     address: args.relay,
                     relayUrl: args.url,
                     transactionFee: args.transactionFee,
                     stake: args.stake,
                     unstakeDelay: args.unstakeDelay
                 }
+                relay.score = this.calculateRelayScore(relay) + this.addScoreRandomness()
```
Comment (alex-forshtat-tbk) [inline src/js/relayclient/ServerHelper.js]: Didn't you move randomness to the "compareRelayScores" ?

File: src/js/relayclient/ServerHelper.js (hunk 2 above)
Comment (drortirosh) [reply src/js/relayclient/ServerHelper.js]: requires too much work... need to save the config params as members to be used by another fucntion..

File: src/js/relayclient/ServerHelper.js (hunk 3 above)
Comment (drortirosh) [reply src/js/relayclient/ServerHelper.js]: oops...

**Pull request merged**